### ISSUE TITLE
Add binary threshold option to "map" display

### ISSUE THREAD

**Issue by roncapat** [ros2/rviz#798]
I'd like to know whether there is interest for an eventual PR (not yet developed) implementing the following:

in the 'map' display:
- add a checkbox enabling/disabling binary map representation;
- add a numeric input filed for specifying a numeric threshold
- if the control box is checked, every cell with cost less than the threshold is colored as selected palette(0), while the other cells are colored as palette(100)
By my current understanding, value 100 is considered obstacle by at least one nav2 planning plugin.

Another missing feature for me is the lack of documentation of the color palettes, which are nowhere plotted with numbered scale. I discovered color for illegal values and other stuff only by looking at the palette code. Any suggestion on how to contribute something useful, or point to some pre-existing documentation that I did not found yet?

**Comment by roncapat** [comment]
#846 still does not solve the issue of not having an easy way to check color scale value correspondence in GUI. I may add a simple optional panel to the GUI with a separate PR in future, if sounds ok.

**Comment by roncapat** [comment]
I may push code for color legend panel after #846 is closed.
See the picture for my current implementation of color legend (automatically updates with selected palette).

Code here: https://github.com/roncapat/rviz/tree/ros2_map_legend
![Screenshot from 2022-04-05 17-23-56](https://user-images.githubusercontent.com/13013812/161789008-0c926685-5c8e-477a-bc14-8680fe16843f.png)

**Comment by roncapat** [comment]
@clalancette any interest in a legend like the one above? Or we close this issue as it's main goal has been achieved by the merged PR?

**Comment by clalancette** [comment]
> @clalancette any interest in a legend like the one above? Or we close this issue as it's main goal has been achieved by the merged PR?

I'm not entirely sure whether we should do a legend or not, but at least the original goal of this issue has been solved.  So I'd say let's close it.

**Comment by roncapat** [comment]
Understood :) Closing
